pdlua.c

```diff
@@ -1088,13 +1112,27 @@ static t_int *pdlua_perform(t_int *w){
             }
         }
         lua_pop(__L(), 1 + o->sigoutlets);
-        return w + o->siginlets + o->sigoutlets + 3;
+        return w + 2 + o->siginlets + o->sigoutlets;
     }
-    
+
     for (int i = o->sigoutlets - 1; i >= 0; i--)
     {
-        t_float *out = (t_float*)w[i + 3 + o->siginlets];
-        for (int j = 0; j < nblock; j++)
+        if (!lua_istable(__L(), -1)) {
+            pd_error(o, "pdlua_perform: expected table for outlet %d", i);
+            lua_pop(__L(), 1);
+            continue;
+        }
```

Comment:
remove with 5053034

Reply:
I'm not sure that's correct. You're using data that's returned by user Lua code there, right? Then you *always* need to check that it's valid.

Reply:
my assumption was that the checks here are enough (like in the current implementation by tim):
https://github.com/agraef/pd-lua/blob/505303451d4189457547760390e1a692c7574ece/pdlua.c#L1101-L1117 

... and there is also the fallback when writing the samples:
https://github.com/agraef/pd-lua/blob/505303451d4189457547760390e1a692c7574ece/pdlua.c#L1131-L1142

Reply:
I don't understand. If I understood you correctly, you're saying that the `!lua_istable(__L(), -1)` check isn't needed anymore, and that this is because it's already done elsewhere? You obviously don't need to check exactly the same value twice, but...

This leaves me even more confused as before. ;-) Maybe we need to take a look at the entire code of `pdlua_perform` in context again to make sure that you didn't butcher it and all the required checks are still there?

Reply:
Ok, I've pulled your branch now and read through the code, looks good to me!

Reply:
In fact I was surprised at how deceptively simple it all looks. It still amazes that this will work as expected no matter what the host version is. Great job!

Reply:
cool - happy to hear! and the solution for handling different host versions is obviously not mine - thanks go to christof and his vstplugin here. :)